### PR TITLE
feature: add helper methods to Quantile and ensure precision safety

### PR DESCRIPTION
**Pull request by lschilders**
This pull request significantly improves the precision and robustness of the `Quantile` type in `openstef_core/types.py` by switching to `Decimal` arithmetic for all conversions and string formatting, and by adding new utility methods. Comprehensive unit tests have been added to ensure correct behavior and to cover edge cases.

**Precision improvements and new features for `Quantile`:**

* Switched all internal arithmetic in `Quantile` to use `Decimal`, ensuring that values like `Quantile(0.999)` round-trip exactly through `format()`, `parse()`, `to_percentile()`, and `from_percentile()` without floating-point rounding errors. [[1]](diffhunk://#diff-a5fdd3bb0fdb7e7fda4a88fd6ed02992f205ca1d03765b5be5ebf87867b04239R17) [[2]](diffhunk://#diff-a5fdd3bb0fdb7e7fda4a88fd6ed02992f205ca1d03765b5be5ebf87867b04239R301-R304) [[3]](diffhunk://#diff-a5fdd3bb0fdb7e7fda4a88fd6ed02992f205ca1d03765b5be5ebf87867b04239R355-R379)
* Added new methods: `from_percentile`, `to_percentile`, and `inverse`, providing convenient and precise conversions between quantiles and percentiles, and to compute the complementary quantile.
* Improved the `format` and `parse` methods to handle arbitrary decimal precision and to use `Decimal` for exact string conversions. [[1]](diffhunk://#diff-a5fdd3bb0fdb7e7fda4a88fd6ed02992f205ca1d03765b5be5ebf87867b04239R355-R379) [[2]](diffhunk://#diff-a5fdd3bb0fdb7e7fda4a88fd6ed02992f205ca1d03765b5be5ebf87867b04239L377-R394)
* Updated the quantile string validation regex to support up to three digits and arbitrary decimal places, e.g., `quantile_P99.99`.

**Testing and validation:**

* Added comprehensive unit tests for all new and updated `Quantile` methods, including round-trip conversions, edge cases, error handling, and the new `inverse` method.

### CODE REVIEW

File: packages/openstef-core/src/openstef_core/types.py [hunk 1]
```diff
@@ -387,9 +403,40 @@ def is_valid_quantile_string(quantile_str: str) -> bool:
...
+
+    def to_percentile(self) -> float:
+        """Convert this Quantile to a percentile value.
+
+        Returns:
+            Percentile value corresponding to this Quantile.
+        """
+        return float(self._percentile_decimal())
+
+    def inverse(self) -> Self:
```
Comment (MvLieshout) [inline packages/openstef-core/src/openstef_core/types.py]: I think the name `inverse` is kind of misleading, as it can also mean 1/q (https://en.wikipedia.org/wiki/Multiplicative_inverse). I think the common name for this is complement (https://en.wikipedia.org/wiki/Complementary_event).

**Review by MvLieshout — COMMENTED** [review]
Looks good. 1 naming suggestion.

File: packages/openstef-core/src/openstef_core/types.py (hunk 1 above)
Comment (lschilders) [reply packages/openstef-core/src/openstef_core/types.py]: good one. will change it.

**Review by MvLieshout — DISMISSED** [review]
LGTM!

**Review by MvLieshout — APPROVED** [review]
